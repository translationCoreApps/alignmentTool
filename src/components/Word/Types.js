export const SECONDARY_WORD = 'secondary';
export const PRIMARY_WORD = 'primary';
