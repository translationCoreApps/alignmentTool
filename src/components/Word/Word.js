import React, { Component } from 'react';
import PropTypes from 'prop-types';
import WordOccurrence from './WordOccurrence';

const defaultStyles = {
  borderLeft: '5px solid #44C6FF',
  padding: '5px',
  backgroundColor: '#FFFFFF',
  boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
  cursor: 'move',
  display: 'flex',
  flexDirection: 'row'
};

/**
 * Renders a single word
 * @property {string} word - the represented word
 * @property {int} occurrence
 * @property {int} occurrences
 */
class Word extends Component {
  render() {
    const { word, occurrence, occurrences, style } = this.props;
    const wordStyle = {...defaultStyles, ...style};
    return (
      <div style={wordStyle}>
        <span style={{flex: 1}}>
          {word}
        </span>
        <WordOccurrence occurrence={occurrence}
                        occurrences={occurrences}/>
      </div>
    );
  }
}

Word.propTypes = {
  style: PropTypes.object,
  word: PropTypes.string.isRequired,
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired
};

Word.defaultProps = {
  style: {}
};

export default Word;
