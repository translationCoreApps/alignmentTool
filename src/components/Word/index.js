import React from 'react';
import PropTypes from 'prop-types';
import WordOccurrence from './WordOccurrence';

const styles = {
  root: {
    borderLeft: '5px solid #44C6FF',
    padding: '10px',
    backgroundColor: '#FFFFFF',
    boxShadow: "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
    cursor: 'move',
    display: 'flex',
    flexDirection: 'row'
  },
  word: {
    cursor: 'pointer'
  }
};

/**
 * Renders a single word
 * @param {string} word - the represented word
 * @param {int} occurrence
 * @param {int} occurrences
 * @param {object} [style] - styles passed  through to the component
 * @param {func} [onClick] - callback when the word is clicked
 * @constructor
 */
const Word = ({word, occurrence, occurrences, style, onClick}) => {
  const rootStyles = {...styles.root, ...style};
  let wordStyles = null;
  if(typeof onClick === 'function') {
    wordStyles = styles.word;
  }
  return (
    <div style={rootStyles}>
      <span style={{flex: 1}}>
        <span onClick={onClick} style={wordStyles}>
          {word}
        </span>
      </span>
      <WordOccurrence occurrence={occurrence}
                      occurrences={occurrences}/>
    </div>
  );
};

Word.propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.object,
  occurrence: PropTypes.number,
  occurrences: PropTypes.number,
  word: PropTypes.string.isRequired
};

Word.defaultProps = {
  style: {},
  occurrence: 1,
  occurrences: 1
};

export default Word;
