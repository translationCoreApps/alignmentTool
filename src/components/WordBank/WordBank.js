import React from 'react';
import PropTypes from 'prop-types';
import SecondaryWord from '../SecondaryWord';

/**
 * Renders a list of words that need to be aligned
 * @param {int} chapter
 * @param {int} verse
 * @param {object} alignmentData
 * @param {bool} isOver
 * @return {*}
 * @constructor
 */
const WordBank = ({
  chapter,
  verse,
  alignmentData,
  isOver,
}) => {
  if (chapter && verse) {
    let wordBank = [];
    if (alignmentData[chapter] && alignmentData[chapter][verse]) {
      wordBank = alignmentData && alignmentData[chapter] &&
      alignmentData[chapter][verse]
        ? alignmentData[chapter][verse].wordBank
        : [];
    }

    if(isOver) {
      return (
        <div style={{
          border: '3px dashed #44C6FF',
          height: '100%',
          width: '100%',
        }}/>
      );
    } else {
      return (
        <React.Fragment>
          {wordBank.map((metadata, index) => (
            <div key={index}
                 style={{margin: '10px'}}>
              <SecondaryWord
                word={metadata.word}
                occurrence={metadata.occurrence}
                occurrences={metadata.occurrences}
              />
            </div>
          ))}
        </React.Fragment>
      );
    }
  }
  return null;
};

WordBank.propTypes = {
  chapter: PropTypes.number,
  verse: PropTypes.number,
  alignmentData: PropTypes.object.isRequired,
  isOver: PropTypes.bool.isRequired
};

export default WordBank;
