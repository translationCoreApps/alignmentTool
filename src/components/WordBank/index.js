import React from 'react';
import PropTypes from 'prop-types';
import { DropTarget } from 'react-dnd';
import ItemTypes from '../ItemTypes';
import WordBank from './WordBank';

/**
 * Renders a word bank with drag-drop support
 */
class DroppableWordBank extends React.Component {
  render() {
    const {chapter, verse, alignmentData, connectDropTarget, isOver} = this.props;

    return connectDropTarget(
      <div>
        <WordBank chapter={chapter}
                  verse={verse}
                  alignmentData={alignmentData}
                  isOver={isOver}/>
      </div>
    );
  }
}

DroppableWordBank.propTypes = {
  chapter: PropTypes.number,
  verse: PropTypes.number,
  alignmentData: PropTypes.object.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  isOver: PropTypes.bool.isRequired,
  moveBackToWordBank: PropTypes.func.isRequired
};

/**
 * Handles drag events on the word bank
 */
const dragHandler = {
  drop (props, monitor) {
    props.moveBackToWordBank(monitor.getItem());
  }
};

const collect = (connect, monitor) => {
  return {
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver(),
    canDrop: monitor.canDrop()
  };
};

export default DropTarget(
  ItemTypes.BOTTOM_WORD,
  dragHandler,
  collect
)(DroppableWordBank);
