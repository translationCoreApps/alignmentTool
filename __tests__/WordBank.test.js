/* eslint-env jest */

jest.unmock('react-dnd');

import React, {Component} from 'react';
import TestBackend from 'react-dnd-test-backend';
import {DragDropContext} from 'react-dnd';
import TestUtils from 'react-dom/test-utils';
import WordBank from '../src/components/WordBank/WordBank';
import Word from '../src/components/Word';
import renderer from 'react-test-renderer';

const alignmentData = {
  '1': require('./fixtures/luke/1.json')
};

describe('snapshot', () => {
  it('has no alignment data', () => {
    const wrapper = renderer.create(
      <WordBank isOver={false}
                alignmentData={{}}
                verse="1"
                chapter="1"/>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has no reference', () => {
    const wrapper = renderer.create(
      <WordBank isOver={false}
                alignmentData={{}}
                verse=""
                chapter=""/>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has data and reference', () => {
    const WrappedWordBank = wrapInTestContext(WordBank);
    const wrapper = renderer.create(
      <WrappedWordBank isOver={false}
                alignmentData={alignmentData}
                verse="1"
                chapter="1"/>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('WordBank', () => {
  let connectDropTarget;

  beforeEach(() => {
    connectDropTarget = div => {
      return div;
    };
  });

  test('renders Luke 1:1', () => {
    // given
    const chapter = '1';
    const verse = '1';
    const expectedWords = alignmentData[chapter][verse].wordBank.length;
    const WordBankContext = wrapInTestContext(WordBank);

    // when
    let root = TestUtils.renderIntoDocument(
      <WordBankContext
        chapter={chapter}
        verse={verse}
        alignmentData={alignmentData}
        connectDropTarget={connectDropTarget}
        isOver={false}
      />,
    );

    // then
    let wordBankItems = TestUtils.scryRenderedComponentsWithType(root, Word);
    expect(wordBankItems).toBeTruthy();
    expect(wordBankItems.length).toEqual(expectedWords);
  });

  test('renders undefined Luke 1:81 without crashing', () => {
    // given
    const expectedWords = 0;
    const WordBankContext = wrapInTestContext(WordBank);

    // when
    let root = TestUtils.renderIntoDocument(
      <WordBankContext
        chapter="1"
        verse="81"
        alignmentData={alignmentData}
        connectDropTarget={connectDropTarget}
        isOver={false}
      />,
    );

    // then
    let wordBankItems = TestUtils.scryRenderedComponentsWithType(root, Word);
    expect(wordBankItems).toBeTruthy();
    expect(wordBankItems.length).toEqual(expectedWords);
  });
});

//
// Helpers
//

/**
 * Wraps a component into a DragDropContext that uses the TestBackend.
 */
function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(
    class TestContextContainer extends Component {
      render() {
        return <DecoratedComponent {...this.props} />;
      }
    },
  );
}
