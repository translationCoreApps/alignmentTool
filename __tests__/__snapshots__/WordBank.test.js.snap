// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot has data and reference 1`] = `
Array [
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            Many
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            have
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          1
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            taken
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            on
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          1
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            the
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          1
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            work
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            of
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          1
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            putting
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            together
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            an
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          1
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            account
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            of
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          2
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            the
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          2
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            things
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            that
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            have
          </span>
        </span>
        <sup
          style={
            Object {
              "marginTop": "2px",
              "opacity": "0.8",
              "top": 0,
            }
          }
        >
          2
        </sup>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            been
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            fulfilled
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            among
          </span>
        </span>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderLeft": "5px solid #44C6FF",
            "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
            "cursor": "move",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "padding": "10px",
          }
        }
      >
        <span
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <span
            onClick={undefined}
            style={null}
          >
            us
          </span>
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`snapshot has no alignment data 1`] = `null`;

exports[`snapshot has no reference 1`] = `null`;
